### ISSUE TITLE
try/catch as error 

### ISSUE THREAD

**Issue by williammartin** [storyscript/runtime#476]
As an end user of a story that doesn't have access to logs, it's really difficult to debug if a story can't handle errors returning from services.

For example:

```
$ curl -vvv -X POST -d '{"document":"", "properties":{}}' -H 'Content-Type: application/json' https://stories.storyscriptapp.com/openapi2omg/convert

...
HTTP 500: Story execution failed
```

while the logs provide the useful error (missing content):

```
Aug 20 15:41:46   ERROR      runtime: An exception has occurred:
Failed to invoke service! Status code: 400; response body: {"error": "No specification parsed, cannot validate!"}
    at line 2: when server listen path:"/openapi2omg/convert" method:"post" as req (in src/omg/openapi.story)
    at line 3: result = openapi2omg convert spec:req.body["document"] properties:req.body["properties"] (in src/omg/openapi.story)
```

I think we need some generic method of allowing storywriters to handle errors coming from services e.g.

```
http server as server
  when server listen path:"/openapi2omg/convert" method:"post" as req
    try
        result = openapi2omg convert spec:req.body["document"] properties:req.body["properties"] 
        req write content:result
    catch as e
        req write content:e
```

We already support `try/catch` but being able to get access to error responses would allow storywriters to:
 - provide useful feedback to callers
 - decide based on the error how to proceed

**Comment by wilzbach** [comment]
See e.g. https://github.com/storyscript/storyscript/issues/954 for prior discussions

tl;dr: we don't want to introduce this until we gather more evidence that it's needed because
- users shouldn't be doing much exception handling
- we want sth. better than try/catch eventually
- there's no easy way to introduce standard error messages into OMG

**Comment by williammartin** [comment]
I also do not like `try/catch` and if we have something better that's great.

I was wondering whether we might need to standardise success/failure in `OMG`.

Fine re: evidence, I suspect we'll see a lot more frustration when the story consumer is not the story author.

Thanks.

**Comment by judepereira** [comment]
@wilzbach So here's why we need it now, or at least a simplified version of it:
1. The tool `openapi2omg` is exposed via a story
2. If `openapi2omg convert` works fine, then we're all good
3. If `openapi2omg convert` results in an error during conversion, it will produce an HTTP 400, along with the response body indicating this error. Now, I would like to forward this error to the caller of the story as well, since they would find the error useful. Today, a StoryscriptError is thrown when a service results in an HTTP 400.

**Comment by TonyRice** [comment]
I think this makes a good case for re-introducing the error within try-catch.
